core/spec/models/spree/wallet_spec.rb

```diff
@@ -6,6 +6,24 @@
   let(:store_credit) { create(:store_credit, user_id: user.id) }
   subject(:wallet) { Spree::Wallet.new(user) }
 
+  describe '.payment_source_helper' do
+    let(:payment_source) { Object.const_set('FakePaymentSource', Class.new(Spree::PaymentSource)) }
+    let(:non_payment_source) { Object.const_set('FakeNonPaymentSource', Class.new) }
+
+    context 'when source is a Spree::PaymentSource' do
+      it 'defines a helper method for the class' do
+        described_class.payment_source_helper payment_source
+        expect(wallet.respond_to?(:fake_payment_sources)).to eql true
```

Comment:
I'd rather replace the `respond_to` expectation with one, which checks the actual contents of the returned collection. I think that `eq` is preferred to `eql`.

Great that you added a test for the ArgumentError.